src/rippled/NFTTransactions.js

```diff
@@ -0,0 +1,43 @@
+import { getNFTTransactions as getNFTTxs } from './lib/rippled'
+import { formatTransaction } from './lib/utils'
+import summarize from './lib/txSummary'
+import logger from './lib/logger'
+
+const log = logger({ name: 'NFT transactions' })
+export const getNFTTransactions = (rippledSocket, tokenId, limit, marker) =>
+  getNFTTxs(rippledSocket, tokenId, limit, marker, undefined)
+    .then((data) => {
+      const transactions = data.transactions.map((tx) => {
+        const txn = formatTransaction(tx)
+        return summarize(txn, true)
+      })
+
+      return {
+        transactions,
+        marker: data.marker,
+      }
+    })
+    .then((d) => d)
+    .catch((error) => {
+      log.error(error.toString())
+      throw error
+    })
```

Comment:
do you prefer this syntax to just `await`? i don't

Reply:
are you talking about then/catch? most other files use this instead of await...

Reply:
doesn't matter - you can use whichever you want. with `await` you have the benefit of  traditional `try` and `catch`